pkg/serverless/metrics/enhanced_metrics_test.go

```diff
@@ -33,42 +27,33 @@ func TestGenerateEnhancedMetricsFromFunctionLogOutOfMemory(t *testing.T) {
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(outOfMemoryLog.Time.UnixNano()),
+		Timestamp:  float64(currentTime.UnixNano()),
 	}})
 }
 
 func TestGenerateEnhancedMetricsFromFunctionLogNoMetric(t *testing.T) {
-	outOfMemoryLog := aws.LogMessage{
-		Type:         aws.LogTypeFunction,
-		StringRecord: "Task timed out after 30.03 seconds",
-		Time:         time.Now(),
-	}
 	metricsChan := make(chan []metrics.MetricSample, 1)
 	tags := []string{"functionname:test-function"}
 
-	go generateEnhancedMetricsFromFunctionLog(outOfMemoryLog, tags, metricsChan)
+	go GenerateEnhancedMetricsFromFunctionLog("Task timed out after 30.03 seconds", time.Now(), tags, metricsChan)
 
 	assert.Equal(t, len(metricsChan), 0)
 }
 
 func TestGenerateEnhancedMetricsFromReportLogColdStart(t *testing.T) {
-	reportLog := aws.LogMessage{
-		Type: aws.LogTypePlatformReport,
-		Time: time.Now(),
-		ObjectRecord: aws.PlatformObjectRecord{
-			Metrics: aws.ReportLogMetrics{
-				DurationMs:       1000.0,
-				BilledDurationMs: 800.0,
-				MemorySizeMB:     1024.0,
-				MaxMemoryUsedMB:  256.0,
-				InitDurationMs:   100.0,
-			},
+	objectRecord := PlatformObjectRecord{
+		Metrics: ReportLogMetrics{
+			DurationMs:       1000.0,
+			BilledDurationMs: 800.0,
+			MemorySizeMB:     1024.0,
+			MaxMemoryUsedMB:  256.0,
+			InitDurationMs:   100.0,
 		},
 	}
 	metricsChan := make(chan []metrics.MetricSample)
 	tags := []string{"functionname:test-function"}
-
-	go generateEnhancedMetricsFromReportLog(reportLog, tags, metricsChan)
+	currentTime := time.Now()
+	go GenerateEnhancedMetricsFromReportLog(objectRecord, currentTime, tags, metricsChan)
```

Comment:
This threw me off a bit because the time used here should be the time from the log message, not the current time. It's just a test, but I would rename the variable to `timestamp` so that it's more obvious that you're not supposed to use the current time with this function.

Reply:
Or maybe `reportLogTime`